src/Swarm/Util.hs

```diff
@@ -134,6 +142,58 @@ cycleEnum e
   | e == maxBound = minBound
   | otherwise = succ e
 
+-- | Drop repeated elements that are adjacent to each other.
+--
+-- >>> uniq []
+-- []
+-- >>> uniq [1..5]
+-- [1,2,3,4,5]
+-- >>> uniq (replicate 10 'a')
+-- "a"
+-- >>> uniq "abbbccd"
+-- "abcd"
+uniq :: Eq a => [a] -> [a]
```

Comment:
Why do we need a quadratic-time `uniq` rather than an n lg n -time version using `Ord`?

Reply:
This should be linear time unix style uniq, unless I made a mistake. 🙂 

It is used on already sorted list.

Reply:
Oh, whoops, that's what I get for reading too quickly. :facepalm: